### ISSUE TITLE
sitemap image URLs do not match with those on product pages

### ISSUE THREAD

**Issue by maderlock** [magento/magento2#7504]
This is currently stopping product images from being correctly indexed by google, which is a serious problem for our clients.

### Preconditions

1. Mageneto 2.1.2

### Steps to reproduce

1. Create a product and add an image.
2. Generate a sitemap

### Expected result

1. img src on product page will match image:loc in sitemap

### Actual result

1. They do not match

The problem here is that google will not index the images from the sitemap as it the URLs do not match up.

However, I have struggled to make these match up. gallery.phtml outputs the image as follows:
````
$imageUrl = $this->helper('Magento\Catalog\Helper\Image')
            ->init($block->getProduct(), 'product_page_image_large')
            ->setImageFile($block->getImageFile())
            ->getUrl();
````

The sitemap gets the image url directly from the product attribute not really consecutive lines, but taken from module-sitemap/Model/Sitemap.php:
````
foreach ($this->_productFactory->create()->getCollection($storeId) as $item) {
$images = $item->getImages();
foreach ($images->getCollection() as $image) {
...
$row .= '<image:loc>' . htmlspecialchars($this->_getMediaUrl($image->getUrl())) . '</image:loc>';
````

The problem here is that $this->_productFactory->create()->getCollection($storeId) gets an array rather than the actual products. Therefore I do not think we can use this with the image helper.

Any thoughts on how to fix this issue?

**Comment by maderlock** [comment]
I'm going to attempt a fix. My current thought is that we'll have to do full product loads, or at least via a collection, so that the image helper can be called. My concern is that this will slow this process down considerably, but at this point it's much more important that it's outputting the correct URLs.

**Comment by Andrii-Deineha** [comment]
@maderlock  Hi, thanks for reporting this issue. Internal ticket was created MAGETWO-70962.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-70056

**Comment by magento-engcom-team** [comment]
@maderlock thank you for your report.
The fix for this issue is already available in release 2.2.0
